docs/guide/xlang_type_mapping.md

```diff
@@ -4,4 +4,76 @@ sidebar_position: 3
+| timestamp          | 18      | Instant         | datetime             | Number          | std::chrono::nanoseconds       | Time             | DateTime         |
+| decimal            | 19      | BigDecimal      | Decimal              | bigint          | /                              | /                | /                |
+| binary             | 20      | byte[]          | bytes                | /               | `int8_t[n]/vector<T>`          | `[n]int8/[]T`    | `Vec<i8>`        |
+| array              | 20      | array           | np.ndarray           | /               | /                              | array/slice      | Vec              |
+| bool_array         | 21      | bool[]          | ndarray(np.bool_)    | /               | `bool[n]`                      | `[n]bool/[]T`    | `Vec<bool>`      |
+| int16_array        | 22      | short[]         | ndarray(int16)       | /               | `int16_t[n]/vector<T>`         | `[n]int16/[]T`   | `Vec<i16>`       |
+| int32_array        | 23      | int[]           | ndarray(int32)       | /               | `int32_t[n]/vector<T>`         | `[n]int32/[]T`   | `Vec<i32>`       |
+| int64_array        | 24      | long[]          | ndarray(int64)       | /               | `int64_t[n]/vector<T>`         | `[n]int64/[]T`   | `Vec<i64>`       |
+| float16_array      | 25      | float[]         | ndarray(float16)     | /               | `fury::float16_t[n]/vector<T>` | `[n]float16/[]T` | `Vec<fury::f16>` |
+| float32_array      | 26      | float[]         | ndarray(float32)     | /               | `float[n]/vector<T>`           | `[n]float32/[]T` | `Vec<f32>`       |
+| float64_array      | 27      | double[]        | ndarray(float64)     | /               | `double[n]/vector<T>`          | `[n]float64/[]T` | `Vec<f64>`       |
+| tensor             | 28      | /               | /                    | /               | /                              | /                | /                |
+| sparse tensor      | 29      | /               | /                    | /               | /                              | /                | /                |
+| arrow record batch | 30      | /               | /                    | /               | /                              | /                | /                |
+| arrow table        | 31      | /               | /                    | /               | /                              | /                | /                |
+
+# Type annotation(not implemented)
+
+Due to differences between type systems of languages, those types can't be mapped one-to-one between languages. Users
+can provide meta hints for fields of a type, or for the whole type. Such information can be provided in other languages
+too:
+
+- java: use annotation.
+- cpp: use macro and template.
+- golang: use struct tag.
+- python: use typehint.
```

Comment:
I have a confusion here, whether decorator can also achieve similar functions.

Reply:
decorator can only be used to decorate method or class, I don't think it can be used here. Descriptor may be suitable, see https://docs.djangoproject.com/en/5.0/ref/models/fields/

Reply:
Great! Thanks for replying to this question.